android/src/main/java/com/reactnativepagerview/PagerViewViewManager.kt

```diff
@@ -25,10 +27,25 @@ class PagerViewViewManager : ViewGroupManager<ViewPager2>() {
     return REACT_CLASS
   }
 
+  fun createFragmentAdapter(reactContext: ThemedReactContext): FragmentAdapter {
+    return FragmentAdapter((reactContext.currentActivity as FragmentActivity?)!!)
+  }
+
   override fun createViewInstance(reactContext: ThemedReactContext): ViewPager2 {
     val vp = ViewPager2(reactContext)
-    val adapter = FragmentAdapter((reactContext.currentActivity as FragmentActivity?)!!)
-    vp.adapter = adapter
+    if(reactContext.hasCurrentActivity()) {
+      vp.adapter = createFragmentAdapter(reactContext);
+    }
+    lifecycleEventListener = object : LifecycleEventListener {
```

Comment:
I wanted to do as you extended, but then I made that without a closure, we don`t have access to reactContext and ViewPager, what do you think about this solution?